data_structures/src/chain.rs

```diff
@@ -2913,10 +2913,64 @@ impl ReputationEngine {
         self.ars
             .active_identities()
             .cloned()
-            .sorted_by_key(|&pkh| std::cmp::Reverse((self.trs.get(&pkh), pkh)))
+            .sorted_by(|a, b| compare_reputed_pkh(a, b, &self).reverse())
             .collect()
     }
 }
+/// Compare 2 PublicKeyHashes comparing:
+/// First: reputation
+/// Second: PublicKeyHash bytes from an offset
+fn compare_reputed_pkh(
+    a: &PublicKeyHash,
+    b: &PublicKeyHash,
+    rep_eng: &ReputationEngine,
+) -> Ordering {
+    let rep_a = rep_eng.trs().get(a).0;
+    let rep_b = rep_eng.trs().get(b).0;
+
+    rep_a.cmp(&rep_b).then_with(|| {
+        compare_vector_with_offset(&a.hash, &b.hash, rep_eng.current_alpha.0 as usize)
+    })
+}
+
+/// Auxiliary function that compares 2 vectors with an offset
+fn compare_vector_with_offset(a: &[u8], b: &[u8], offset: usize) -> Ordering {
```

Comment:
I think this function should be discussed in a separated PR. It would be great if you could create an issue and change the body of this function to something super simple like

```rust
    // FIXME(#1506): implement better tie breaker
    a.cmp(b)
```

This way we can clearly state the design goals and the attack vectors.

I prefer to remove the magic function because I wrote a short script and with 1 hour of proof of work I was able to generate a PKH that is better than 77% of the random PKHs, so I assume that this function can be easily broken: [Playground](https://play.rust-lang.org/?version=stable&mode=release&edition=2018&gist=5de2247ee012c06975927ff7bf5fd580)